### DISCUSSION TITLE
Relay Dependency

### DISCUSSION THREAD

**Discussion by Incredible-IT** [theyosh/TerrariumPI#934] [General]
Hi,
Thanks for such a wonderful system. 
I have a Hutch with 3 separate enclosures. Up to now I have been manually controlling the humidity by watering the substrate once or so a week. One of my habitats actually has a rain system that is running once a day for 10-15 sec just to keep the humidity up. 

I now want to change things up a bit. I want to branch out of my one water container with one pump and control the humidity for the 3 separate habitats with solenoid valves. To do this I was thinking of having one relay turn on the pump and another turn on the solenoid. So humidity in habitat 1 goes low, Relay 1 for the pump turns on, and Relay2 for habitat 1's rain. Humidity in Habitat 2 goes low Relay 1 for the pump turns on and Relay 3 for Habitat 2 rain comes on. Same deal for Habitat 3. I want to use one relay for pump all 3 enclosures. Is this possible or do you have another suggestion?

**Comment by theyosh** [comment]
I think this should work for you: https://theyosh.github.io/TerrariumPI/faq/how-to-use-a-relay-multiple-times/

**Reply by Incredible-IT** [reply]
Yes thank you. I found that a couple days ago and was just coming back to post the answer.

However, I now have a problem where the pump relay (duplicated) stays on. shouldn't it turn of along with the solenoid relay?

**Reply by theyosh** [reply]
The relays are not depending on each other, but rather combined through an area. So I expect you have one area with the solenoid relay and the duplicated relay.
That should work.

But for testing, do not add the relay to an area, so you can just manually toggle it and see if the water pump reacts correctly.

Also, when you have duplicated multiple times the same relay, I cannot tell what will happen when 3 areas are controlling the same water pump relay though duplicated relays.

**Reply by Incredible-IT** [reply]
Thanks for the quick responses. Another question along the same lines. 

When humidity drops below the alarm level. I want the mist to come on for 15 sec. You showed me that before. That should be enough to bring the humidity up. I would like for it to wait 10 min or so to settle out. Is that the settle setting? Is that also seconds?

**Reply by Incredible-IT** [reply]
> I think this should work for you: https://theyosh.github.io/TerrariumPI/faq/how-to-use-a-relay-multiple-times/

Ok, This does kind of work. It will toggle on the pump but it does not toggle off the pump.

To test on one enclosure, I tried it with the original Pump Relay and the Solenoid Relay. RPump and RSel-1 When the Humidity drops below alarm both relays turn on for 15 sec. I have a "settle" time of 600sec. This setup seems to work great for the one enclosure. Both relays turn on and off as they are supposed to. If the humidity is not above the alarm threshold, the relays kick on for 15 sec again. Then they both turn off.

To further the test, still on one enclosure, I disassociated the Original Relay for the pump with all areas. I used the above instructions to duplicate the Pump Relay. I called it RS-1. (relay script) I then setup the same enclosure with the RS-1 and the RSel-1. It works as expected except RS-1 does not turn off the RPump relay. The Solenoid Relay does turn off, but not the duplicate RS-1 of the Pump Relay.

Any suggestions?

**Reply by theyosh** [reply]
In order to do proper testing, please do not create areas. That will give more noise on the conversation. Also when a relays is assigned in an area, the area will always turn the relays at the needed state according to the area logic.

So, first you have added the original pump relay.
Than you have created a duplicated relay and added to the system.

So with only those 2 relays added, there is nothing that can interfere the testing. So make sure you have removed all relays and enclosures with all areas. Else it will get complicated to explain and do proper testing.

So toggle the duplicated script relay, and see what happens. This is just a manual action. If that does not work, there is no use for creating areas. Because the relays is already not working, and an area will never fix that.

**Comment by Incredible-IT** [comment]
I already tested it that way per your suggestion. When I first setup and tested the duplicate relay, before adding it to the enclosure, I did test it manually in relays. It does work correctly there. When I toggle the Script Relay it toggles on the actual relay, and off again when I toggle off.
So that parts was tested before I added to an enclosure. 
When I add to the enclosure as just the real pump relay and the solenoid relay everything works as expected. Both relays toggle on/off as they should. 
When I add the Script Relay and the Solenoid Relay to the Enclosure, both toggle on for the 15sec. The Solenoid Relay toggles off but the Script Relay does not turn off the pump.

**To keep things from getting complicated I do delete the "area" after my test. I don't want it turning on while I am not there. Also start with fresh settings for a test.

I was wondering, I noticed when adding more than one relay, we have the options on, each relay, for delay on and delay off. They are coded for minutes, not sec. When dealing with a water pump, a lot can happen in a min. Is there a way, maybe behind the scenes, to set that delay in seconds? Maybe setting an off delay there would force the toggle?

**Reply by theyosh** [reply]
Set all delays for all relays to `0`. That is not what you need. Could else post a screenshot of your area setup? Than I can maybe easier see how you made the setup. 

Because you want to use the fields `Begin time`, `End time`, `Power on time` and `Settle time`: https://theyosh.github.io/TerrariumPI/scenarios/#timer-wizard

And if you only want to run `once a day` for `15 seconds` you can just set the Begin time on 13:30 and end time to 13:31.
And put the `Power on time` to 15 (seconds) and `Settle time` to 45 (or more seconds).

![image](https://github.com/user-attachments/assets/24011117-c70f-4df8-9739-71c676148175)

**Reply by theyosh** [reply]
And after some testing, my duplicated relay and original relay work like expected:
![image](https://github.com/user-attachments/assets/0fce5c58-289c-4d55-b24d-cb60c2a88b92)

I only tested with one relay in my area:
![image](https://github.com/user-attachments/assets/6d9b5ed4-9718-403f-bfba-74be57b215cd)

**Comment by Incredible-IT** [comment]
I am testing with only one enclosure at the moment. I start with just the plain Humidity Sensor. Alarms are 60-80%. I have the two relays. RDC-2_R-Rain (Solenoid) and RS-2_Pump (Guess). 20 sec on. 40 sec settle. No delays. High Alarm is set to ignore.

![image](https://github.com/user-attachments/assets/18fb6ee3-788c-4be6-857e-b28a3b36fafb)

I'm going to try a Timer scenario. See how that works.

**Comment by Incredible-IT** [comment]
I tried a timer with the rain solenoid relay. and duplicate pump relay. I set it for 8:45P to 8:46P on for 20 sec and off for 40 sec. Both relays turned on and off on time as they should. 

So, at this point, with the test so far, the sensor based trigger has a problem turning off the duplicated relay. Remember it does turn off the real relay properly.

**Comment by Incredible-IT** [comment]
Well that didn't work. It seems that if there are a solenoid relay and a duplicate script relay on the same area it will not stop the scripted duplicate relay. It just keeps running.

I currently have a timer situation going on. This is not what I want but it seems to be working. I would rather the pump and solenoid relays be triggered together from the humidity sensor low alarm.

What I have done is created a separate area for the pump. I have the pump relay on a timer running for 3 min. Then I have my 3 enclosures setup with each solenoid relay scheduled 1 min apart with 30 sec on and 30 sec settle time.

**Comment by Incredible-IT** [comment]
I figured it out...

I tied the signal from each of the solenoid relays to the signal of the pump relay with a diode on each one. Now I only have to trigger the one Solenoid Relay with the Enclosure Area.

When one of the Humidity Alarms is Low, it will activate the rain solenoid for 20 sec. I have a 600 sec settle time so the Alarm will not trigger the solenoid again for 10 min. If the Alarm is still Low the rain will then trigger again.

The diodes keep the signal from back activating the other relays. Since it is only triggered with a ground, no other components needed.

![image](https://github.com/user-attachments/assets/a21c0882-206f-4ee3-b051-53018a2504de)

**Comment by Incredible-IT** [comment]
Update. While this appeared to work correctly. The pump would occasionally stay on. So, I decided to try a pull-up resistor on the pump. I attached a 10K resistor to the pump relay signal. I just did it a little while ago so I will update later this week.

![image](https://github.com/user-attachments/assets/46d707a5-604e-41c0-b9a9-00c89814298d)

**Comment by Incredible-IT** [comment]
Working really well. I think the original problem, with the duplicate relay script, would have been solved with a pullup resistor.
